pkg/staticnetworkconfig/generator.go

```diff
@@ -35,61 +38,85 @@ func New(log logrus.FieldLogger) StaticNetworkConfig {
-	hostsConfig := strings.Split(hostsYAMLS, staticNetworkConfigHostDelimeter)
+	hostsConfig := strings.Split(hostsYAMLS, staticNetworkConfigHostsDelimeter)
 	s.log.Infof("Start configuring static network for %d hosts", len(hostsConfig))
-	executer := &executer.CommonExecuter{}
 	filesList := []StaticNetworkConfigData{}
-	for _, hostConfig := range hostsConfig {
-		f, err := executer.TempFile("", "host-config")
-		if err != nil {
-			s.log.WithError(err).Errorf("Failed to create temp file")
-			return nil, err
-		}
-		_, err = f.WriteString(hostConfig)
-		if err != nil {
-			s.log.WithError(err).Errorf("Failed to write host config to temp file")
-			return nil, err
-		}
-		f.Close()
-		stdout, stderr, retCode := executer.Execute("nmstatectl", "gc", f.Name())
-		if retCode != 0 {
-			msg := fmt.Sprintf("<nmstatectl gc> failed, errorCode %d, stderr %s, input yaml <%s>", retCode, stderr, hostConfig)
-			s.log.Errorf("%s", msg)
-			return nil, fmt.Errorf("%s", msg)
-		}
-		err = s.createNMConnectionFiles(stdout, &filesList)
+	for i, hostConfig := range hostsConfig {
+		hostFileList, err := s.generateHostStaticNetworkConfigData(hostConfig, fmt.Sprintf("host%d", i))
 		if err != nil {
-			s.log.WithError(err).Errorf("failed to create NM connection files")
+			s.log.WithError(err).Errorf("Failed to create static config for host")
 			return nil, err
 		}
-		os.Remove(f.Name())
+		filesList = append(filesList, hostFileList...)
+	}
+	return filesList, nil
+}
+
+func (s *StaticNetworkConfigGenerator) generateHostStaticNetworkConfigData(hostConfigString, hostDir string) ([]StaticNetworkConfigData, error) {
+	hostConfig := strings.Split(hostConfigString, hostStaticNetworkDelimeter)
+	if len(hostConfig) != 2 {
+		msg := fmt.Sprintf("Invalid format of the host config string %s", hostConfig)
+		s.log.Errorf("%s", msg)
+		return nil, fmt.Errorf("%s", msg)
+	}
+	hostYAML := hostConfig[0]
+	macInterfaceMapping := hostConfig[1]
+	executer := &executer.CommonExecuter{}
+	f, err := executer.TempFile("", "host-config")
+	if err != nil {
+		s.log.WithError(err).Errorf("Failed to create temp file")
+		return nil, err
+	}
+	_, err = f.WriteString(hostYAML)
+	if err != nil {
+		s.log.WithError(err).Errorf("Failed to write host config to temp file")
+		return nil, err
 	}
+	f.Close()
+	stdout, stderr, retCode := executer.Execute("nmstatectl", "gc", f.Name())
+	if retCode != 0 {
+		msg := fmt.Sprintf("<nmstatectl gc> failed, errorCode %d, stderr %s, input yaml <%s>", retCode, stderr, hostYAML)
+		s.log.Errorf("%s", msg)
+		return nil, fmt.Errorf("%s", msg)
+	}
+	filesList, err := s.createNMConnectionFiles(stdout, hostDir)
+	if err != nil {
+		s.log.WithError(err).Errorf("failed to create NM connection files")
+		return nil, err
+	}
+	os.Remove(f.Name())
+	mapConfigData := StaticNetworkConfigData{
+		FilePath:     filepath.Join(hostDir, "mac_interface.csv"),
```

Comment:
since the delimiter between attributes is "=", csv (comma separated) isn't the suffix for it.
either 'ini', 'properties' .

Reply:
fixed